### PR TITLE
[Billing RP] Get-AzureRmEnrollmentAccount

### PR DESCRIPTION
**Pull request by wilcobmsft**
## Description

Introducing Get-AzureRmEnrollmentAccount, a cmdlet to let you list EA enrollment accounts you either own or have been granted access to via RBAC.

## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
      cmdlet review [here](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/25)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Billing/AzureRM.Billing.psd1 [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '0.14.0'
+ModuleVersion = '0.15.0'
```
Comment (markcowl) [inline src/ResourceManager/Billing/AzureRM.Billing.psd1]: this is handled by tooling, no need to change the version

File: src/ResourceManager/Billing/ChangeLog.md [hunk 2]
```diff
@@ -19,6 +19,10 @@
 -->
 ## Current Release
 
+## Version 0.15.0
```
Comment (markcowl) [inline src/ResourceManager/Billing/ChangeLog.md]: Please remove the version heading, these changes should just be in 'Current Release'

File: src/ResourceManager/Billing/Commands.Billing.Test/ScenarioTests/EnrollmentAccountTests.ps1 [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.SYNOPSIS
+List billing periods
+#>
+function Test-ListEnrollmentAccounts
+{
+    $enrollmentAccounts = Get-AzureRmEnrollmentAccount
```
Comment (markcowl) [inline src/ResourceManager/Billing/Commands.Billing.Test/ScenarioTests/EnrollmentAccountTests.ps1]: The setup that is necessary to run the test should be part of the test.

File: src/ResourceManager/Billing/Commands.Billing.Test/ScenarioTests/EnrollmentAccountTests.ps1 [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	Assert-NotNull $enrollmentAccounts[0].PrincipalName
+}
+
+<#
+.SYNOPSIS
+Get billing period with specified name
+#>
+function Test-GetEnrollmentAccountWithName
+{
+	$enrollmentAccountObjectId = "c8a9f59a-2d9b-4086-91c8-8988cae3bec3"
```
Comment (markcowl) [inline src/ResourceManager/Billing/Commands.Billing.Test/ScenarioTests/EnrollmentAccountTests.ps1]: Please do not use hard=code values, and ensure that the test setup is part of the test

File: src/ResourceManager/Billing/Commands.Billing.Test/SessionRecords/Microsoft.Azure.Commands.Billing.Test.ScenarioTests.BillingPeriodsTests/TestGetBillingPeriodWithName.json [hunk 5]
```diff
@@ -1,8 +1,8 @@
 {
   "Entries": [
     {
-      "RequestUri": "/subscriptions/0347d63a-421a-43a7-836a-5f389c79cd07/providers/Microsoft.Billing/billingPeriods/201705-1?api-version=2017-04-24-preview",
```
Comment (markcowl) [inline src/ResourceManager/Billing/Commands.Billing.Test/SessionRecords/Microsoft.Azure.Commands.Billing.Test.ScenarioTests.BillingPeriodsTests/TestGetBillingPeriodWithName.json]: Did you change this manually, or did you re-record?

File: src/ResourceManager/Billing/Commands.Billing.Test/SessionRecords/Microsoft.Azure.Commands.Billing.Test.ScenarioTests.BillingPeriodsTests/TestGetBillingPeriodWithNames.json [hunk 6]
```diff
@@ -1,8 +1,8 @@
 {
   "Entries": [
     {
-      "RequestUri": "/subscriptions/0347d63a-421a-43a7-836a-5f389c79cd07/providers/Microsoft.Billing/billingPeriods/201705-1?api-version=2017-04-24-preview",
```
Comment (markcowl) [inline src/ResourceManager/Billing/Commands.Billing.Test/SessionRecords/Microsoft.Azure.Commands.Billing.Test.ScenarioTests.BillingPeriodsTests/TestGetBillingPeriodWithNames.json]: same question on all of these updated recordings - did you re-record, or did you edit the recording?

File: src/ResourceManager/Billing/Commands.Billing/Cmdlets/EnrollmentAccounts/GetAzureRmEnrollmentAccount.cs [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Billing.Cmdlets.EnrollmentAccounts
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmEnrollmentAccount", DefaultParameterSetName = Constants.ParameterSetNames.ListParameterSet), OutputType(typeof(List<PSBillingPeriod>), typeof(PSBillingPeriod))]
+    public class GetAzureRmEnrollmentAccount : AzureBillingCmdletBase
+    {
+        [Parameter(Mandatory = true, HelpMessage = "ObjectId of the enrollment account to get.", ParameterSetName = Constants.ParameterSetNames.SingleItemParameterSet)]
```
Comment (markcowl) [inline src/ResourceManager/Billing/Commands.Billing/Cmdlets/EnrollmentAccounts/GetAzureRmEnrollmentAccount.cs]: Please include Position=0, it should be ok for the customer to enter the object id without using '-ObjectId'

File: src/ResourceManager/Billing/Commands.Billing/Microsoft.Azure.Commands.Billing.Format.ps1xml [hunk 8]
```diff
@@ -81,5 +81,27 @@
         </ListEntries>
       </ListControl>
     </View>
+    <View>
+      <Name>Microsoft.Azure.Commands.Billing.Models.PSEnrollmentAccount</Name>
+      <ViewSelectedBy>
+        <TypeName>Microsoft.Azure.Commands.Billing.Models.PSEnrollmentAccount</TypeName>
+      </ViewSelectedBy>
+      <ListControl>
```
Comment (markcowl) [inline src/ResourceManager/Billing/Commands.Billing/Microsoft.Azure.Commands.Billing.Format.ps1xml]: The default should be a table control.  Please provide a table control format here

File: src/ResourceManager/Billing/Commands.Billing/Properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -46,5 +46,5 @@
 // You can specify all the values or you can default the Build and Revision Numbers 
 // by using the '*' as shown below:
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyVersion("0.14.0")]
-[assembly: AssemblyFileVersion("0.14.0")]
+[assembly: AssemblyVersion("0.15.0")]
```
Comment (markcowl) [inline src/ResourceManager/Billing/Commands.Billing/Properties/AssemblyInfo.cs]: no need to change

File: src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmEnrollmentAccount.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+## SYNTAX
+
+### List (Default)
+```
+Get-AzureRmEnrollmentAccount [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
+```
+
+### Single
+```
+Get-AzureRmEnrollmentAccount -Name <System.String>
```
Comment (markcowl) [inline src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmEnrollmentAccount.md]: this is not accurate - the cmdlet has an ObjectId paraeter, but no name

File: src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmEnrollmentAccount.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+## DESCRIPTION
+The **Get-AzureRmEnrollmentAccount** cmdlet gets enrollment accounts.
+
+## EXAMPLES
+
+### Example 1
+```
+PS C:\> Get-AzureRmEnrollmentAccount
+```
```
Comment (markcowl) [inline src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmEnrollmentAccount.md]: Please include output in the examples

File: src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmEnrollmentAccount.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+Get all available enrollment accounts.
+
+### Example 2
+```
+PS C:\> Get-AzureRmEnrollmentAccount -Name 91825d71-02bd-46e2-95e2-b26369ade34b
```
Comment (markcowl) [inline src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmEnrollmentAccount.md]: ObjectId

File: src/ResourceManager/Billing/AzureRM.Billing.psd1 (hunk 1 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/AzureRM.Billing.psd1]: Reverted

File: src/ResourceManager/Billing/ChangeLog.md (hunk 2 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/ChangeLog.md]: Fixed

File: src/ResourceManager/Billing/Commands.Billing.Test/ScenarioTests/EnrollmentAccountTests.ps1 (hunk 3 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/Commands.Billing.Test/ScenarioTests/EnrollmentAccountTests.ps1]: Unfortunately there's no simple way to create enrollment accounts via PowerShell. These currently have to be created via the EA portal, and this is something a department admin would need to do. Additionally, the owner of the enrollment account will need to accept the invitation to complete the workflow.

I'll go ahead and document the steps required in the test, but I don't believe it's possible for us to automate this at this time.

File: src/ResourceManager/Billing/Commands.Billing.Test/ScenarioTests/EnrollmentAccountTests.ps1 (hunk 4 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/Commands.Billing.Test/ScenarioTests/EnrollmentAccountTests.ps1]: Fixed

File: src/ResourceManager/Billing/Commands.Billing.Test/SessionRecords/Microsoft.Azure.Commands.Billing.Test.ScenarioTests.BillingPeriodsTests/TestGetBillingPeriodWithName.json (hunk 5 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/Commands.Billing.Test/SessionRecords/Microsoft.Azure.Commands.Billing.Test.ScenarioTests.BillingPeriodsTests/TestGetBillingPeriodWithName.json]: I updated the recordings. I'm investigating why recording isn't working for me, and will update the recordings once I have it working.

File: src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmEnrollmentAccount.md (hunk 10 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmEnrollmentAccount.md]: Fixed

File: src/ResourceManager/Billing/Commands.Billing/Properties/AssemblyInfo.cs (hunk 9 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/Commands.Billing/Properties/AssemblyInfo.cs]: Reverted

File: src/ResourceManager/Billing/Commands.Billing/Microsoft.Azure.Commands.Billing.Format.ps1xml (hunk 8 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/Commands.Billing/Microsoft.Azure.Commands.Billing.Format.ps1xml]: Fixed

File: src/ResourceManager/Billing/Commands.Billing/Cmdlets/EnrollmentAccounts/GetAzureRmEnrollmentAccount.cs (hunk 7 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/Commands.Billing/Cmdlets/EnrollmentAccounts/GetAzureRmEnrollmentAccount.cs]: Fixed

File: src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmEnrollmentAccount.md (hunk 11 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmEnrollmentAccount.md]: Done

File: src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmEnrollmentAccount.md (hunk 12 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/Commands.Billing/help/Get-AzureRmEnrollmentAccount.md]: Fixed

File: src/ResourceManager/Billing/Commands.Billing.Test/SessionRecords/Microsoft.Azure.Commands.Billing.Test.ScenarioTests.BillingPeriodsTests/TestGetBillingPeriodWithNames.json (hunk 6 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/Commands.Billing.Test/SessionRecords/Microsoft.Azure.Commands.Billing.Test.ScenarioTests.BillingPeriodsTests/TestGetBillingPeriodWithNames.json]: Got recording to work. All have been re-recorded now.

File: src/ResourceManager/Billing/Commands.Billing.Test/SessionRecords/Microsoft.Azure.Commands.Billing.Test.ScenarioTests.BillingPeriodsTests/TestGetBillingPeriodWithName.json (hunk 5 above)
Comment (wilcobmsft) [reply src/ResourceManager/Billing/Commands.Billing.Test/SessionRecords/Microsoft.Azure.Commands.Billing.Test.ScenarioTests.BillingPeriodsTests/TestGetBillingPeriodWithName.json]: Re-recorded everything.

File: src/ResourceManager/Billing/Commands.Billing.Test/ScenarioTests/EnrollmentAccountTests.ps1 (hunk 3 above)
Comment (markcowl) [reply src/ResourceManager/Billing/Commands.Billing.Test/ScenarioTests/EnrollmentAccountTests.ps1]: Thanks, the comments will help

File: src/ResourceManager/Billing/Commands.Billing/Microsoft.Azure.Commands.Billing.Format.ps1xml [hunk 13]
```diff
@@ -81,5 +81,29 @@
...
+      <TableControl>
+        <TableHeaders>
+          <TableColumnHeader/>
+          <TableColumnHeader/>
+        </TableHeaders>
+        <TableRowEntries>
+          <TableRowEntry>
+            <TableColumnItems>
+              <TableColumnItem>
+                <PropertyName>ObjectId</PropertyName>
```
Comment (markcowl) [inline src/ResourceManager/Billing/Commands.Billing/Microsoft.Azure.Commands.Billing.Format.ps1xml]: nice, thanks

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/686/

**Pull request merged**